snuba/datasets/discover.py

```diff
@@ -263,47 +190,39 @@ def __init__(self) -> None:
             ]
         )
 
+        # TODO: Move the storage definition out of events and transactions dataset so
+        # we can reuse the class itself and avoid the isinstance call.
+        events_dataset = get_dataset(EVENTS)
+        assert isinstance(events_dataset, EventsDataset)
+        transactions_dataset = get_dataset(TRANSACTIONS)
+        assert isinstance(transactions_dataset, TransactionsDataset)
+        storage_selector = DiscoverQueryStorageSelector(
+            events_table=events_dataset.get_storage(),
+            transactions_table=transactions_dataset.get_storage(),
+        )
+
         super().__init__(
-            dataset_schemas=DatasetSchemas(
-                read_schema=DiscoverSchema(
-                    table_source=None,
-                    common_columns=self.__common_columns,
-                    events_columns=self.__events_columns,
-                    transactions_columns=self.__transactions_columns,
-                ),
-                write_schema=None,
-            ),
+            storages=[
+                events_dataset.get_storage(),
+                transactions_dataset.get_storage(),
+            ],
+            storage_selector=storage_selector,
+            abstract_column_set=self.__common_columns
+            + self.__events_columns
+            + self.__transactions_columns,
```

Comment:
If you parenthesize the right hand side of this expression, that will trick black into indenting it and making it easier to read.

Reply:
now I get why that sometimes works and others doesn't